habitat/datasets/pointnav/generator.py

```diff
@@ -0,0 +1,142 @@
+from typing import Optional, List
+
+import numpy as np
+
+from habitat.core.simulator import ShortestPathPoint
+from habitat.tasks.nav.nav_task import NavigationGoal, NavigationEpisode
+from habitat.tasks.nav.shortest_path_follower import ShortestPathFollower
+
+GEODESIC_TO_EUCLID_RATIO_THRESHOLD = 1.1
+NUMBER_RETRIES_PER_TARGET = 10
+NEAR_DIST_LIMIT = 1
+FAR_DIST_LIMIT = 30
+ISLAND_RADIUS_LIMIT = 1.5
+
+
+def _ratio_sample_rate(ratio):
+    """
+
+    :param ratio: geodesic distance ratio to Euclid distance
+    :return: value between 0.008 and 0.144 for ration 1 and 1.1
+    """
+    return 20 * (ratio - 0.98) ** 2
+
+
+def is_compatible_episode(s, t, env, near_dist, far_dist):
+    substract_vec = [a_i - b_i for a_i, b_i in zip(s, t)]
```

Comment:
Why not just `s - t` or `np.array(s) - np.array(t)`?

Reply:
@erikwijmans, `s - t` won't work on python lists, while `np.array(s) - np.array(t)` is good idea.

Reply:
`s` and `t` are gotten from `sample_navigable_point()` which returns `np.array`s

Reply:
In HAPI sim wrapper we convert to list already: https://github.com/facebookresearch/habitat-api/blob/8c5329ce9f777395b45c2daac135377e629ec148/habitat/sims/habitat_simulator.py#L338. Created issue https://github.com/facebookresearch/habitat-api/issues/84 to start moving all HAPI interfaces to np.array.